File: 程式設計：資料結構答案檔/Python/403.py
n, v, threshold = map(int, input().split())
visited = [False for i in range(n+1)]
node = [[] for i in range(n+1)]

for i in range(n-1):
    v0, v1, w = map(int, input().split())
    node[v0].append([v1, w])
    node[v1].append([v0, w])

ans = 0

q = []
visited[v] = True
q.append([v, 2147483647])

while len(q):
    t = q.pop(0)
    
    for first, second in node[t[0]]:
        if visited[first]: continue

        visited[first] = True
        if min([t[1], second]) >= threshold:
               ans+= 1
               q.append([first,  min([t[1], second])])

print(ans)

File: 程式設計：資料結構答案檔/C++/403.cpp
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

int main(void)
{
	int n, v, threshold;
	int v0, v1, w;
	queue<pair<int, int>> q;
	int ans = 0;

	cin >> n >> v >> threshold;
	bool visited[n + 1] = {};
	vector<pair<int, int>> node[n + 1];

	for (int i = 0; i < n - 1; i++)
	{
		cin >> v0 >> v1 >> w;
		node[v0].push_back({v1, w});
		node[v1].push_back({v0, w});
	}

	visited[v] = true;
	q.push({v, 2147483647});
	while (!q.empty())
	{
		pair<int, int> t = q.front();
		q.pop();

		for (auto p : node[t.first])
		{
			if (visited[p.first])
				continue;

			visited[p.first] = true;
			if (min(t.second, p.second) >= threshold)
			{
				ans++;
				q.push({p.first, min(t.second, p.second)});
			}
		}
	}
	cout << ans;

	return 0;
}
